src/ajax.ts

```diff
@@ -37,7 +52,31 @@ function getConnector() {
  * @returns {Promise<any>}
  */
 export function send(url: string, data : any = {}, method = 'get', expectedDataType = 'json'): Promise<any> {
-  return getConnector().then((c) => c.send(url, data, method, expectedDataType));
+  if (method === 'get' || method === 'head') {
+    data = encodeParams(data); //encode in url
+    if (data) {
+      url += (/\?/.test(url) ? '&' : '?') + data;
+    }
+    data = null;
```

Comment:
the logic is, if it is the method is head or get you have to encode the body in the url and provide no real body. Please suggest a better structure